src/game/game.cpp

```diff
@@ -6573,67 +6579,50 @@ void Game::checkImbuements()
 	g_scheduler.addEvent(createSchedulerTask(EVENT_IMBUEMENTINTERVAL, std::bind(&Game::checkImbuements, this)));
 
 	size_t bucket = (lastImbuedBucket + 1) % EVENT_IMBUEMENT_BUCKETS;
+	std::list<Item*> items = imbuedItems[bucket];
 
-	auto it = imbuedItems[bucket].begin(), end = imbuedItems[bucket].end();
-	while (it != end) {
+	for (auto it = items.begin(); it != items.end(); ++it) {
 		Item* item = *it;
 		if (!item) {
 			continue;
 		}
 
 		Player* player = item->getHoldingPlayer();
-		if (!player) {
-			ReleaseItem(item);
-			it = imbuedItems[bucket].erase(it);
+		if (item->isRemoved() || !player) {
```

Comment:
move the item condition to above
```suggestion
		if (!player) {
```